### PR TITLE
Added support for Core ML models

### PR DESCRIPTION
**Pull request by ifielker**
Added support for Core ML models

### CODE REVIEW

File: src/machine-learning/machine-learning-api-client.ts [hunk 1]
```diff
@@ -54,13 +54,23 @@ export interface AutoMLTfliteModelOptions extends ModelOptionsBase {
...
+  };
+}
+export type ModelOptions = ModelOptionsBase | GcsTfliteModelOptions | AutoMLTfliteModelOptions | GcsCoremlModelOptions;
 export type ModelUpdateOptions = ModelOptions & { state?: { published?: boolean }};
 
 export function isGcsTfliteModelOptions(options: ModelOptions): options is GcsTfliteModelOptions {
   const gcsUri = (options as GcsTfliteModelOptions)?.tfliteModel?.gcsTfliteUri;
   return typeof gcsUri !== 'undefined'
 }
+export function isGcsCoremlModelOptions(options: ModelOptions): options is GcsCoremlModelOptions {
```
Comment (hiranya911) [inline src/machine-learning/machine-learning-api-client.ts:69]: Please add spacing around declarations. Definitely around all functions, and ideally interfaces too.

File: test/unit/machine-learning/machine-learning.spec.ts [hunk 2]
```diff
@@ -123,6 +138,67 @@ describe('MachineLearning', () => {
...
+    displayName: 'model_3',
+    tags: ['tag_3', 'tag_4'],
+    state: { published: true },
+    coremlModel: {
+      managedUpload: true,
+      sizeBytes: 33300333,
+    },
+  };
+
+  const MODEL_RESPONSE8: any = {
```
Comment (hiranya911) [inline test/unit/machine-learning/machine-learning.spec.ts:186]: These names are becoming quite hard to follow. It was manageable when it was just 2-3, but now we are up to 8. Going through the test cases, it requires quite a bit of effort to follow and understand the logic. Highly recommend we give more descriptive names to these, so when they appear in the test cases it makes sense.
Suggestions: `COREML_MODEL_WITH_UNKNOWN_FIELD`, `COREML_MODEL_WITH_MANAGED_UPLOAD`, `COREML_MODEL_WITH_GCS_URI` and so on.

File: test/unit/machine-learning/machine-learning-api-client.spec.ts [hunk 3]
```diff
@@ -486,6 +578,32 @@ describe('MachineLearningApiClient', () => {
         });
     });
 
+    it('should resolve properly with tflite managed model', () => {
```
Comment (hiranya911) [inline test/unit/machine-learning/machine-learning-api-client.spec.ts:581]: Drop "properly". Here and below. Also give more context "should resolve with Model for tflite managed model response"

**Review by hiranya911 — CHANGES_REQUESTED** [review]
Looks pretty good. Just a few readability nits.

File: src/machine-learning/machine-learning-api-client.ts (hunk 1 above)
Comment (ifielker) [reply src/machine-learning/machine-learning-api-client.ts:69]: done

File: test/unit/machine-learning/machine-learning.spec.ts (hunk 2 above)
Comment (ifielker) [reply test/unit/machine-learning/machine-learning.spec.ts:186]: done

File: test/unit/machine-learning/machine-learning-api-client.spec.ts (hunk 3 above)
Comment (ifielker) [reply test/unit/machine-learning/machine-learning-api-client.spec.ts:581]: done

**Review by hiranya911 — APPROVED** [review]
Thanks for making the changes. LGTM 👍
